homeassistant/components/home_connect/switch.py

```diff
@@ -35,18 +63,83 @@ async def async_setup_entry(
     def get_entities():
         """Get a list of entities."""
         entities = []
-        hc_api = hass.data[DOMAIN][config_entry.entry_id]
+        hc_api: ConfigEntryAuth = hass.data[DOMAIN][config_entry.entry_id]
         for device_dict in hc_api.devices:
             entity_dicts = device_dict.get(CONF_ENTITIES, {}).get("switch", [])
             entity_list = [HomeConnectProgramSwitch(**d) for d in entity_dicts]
             entity_list += [HomeConnectPowerSwitch(device_dict[CONF_DEVICE])]
             entity_list += [HomeConnectChildLockSwitch(device_dict[CONF_DEVICE])]
+            # Auto-discover entities
+            hc_device: HomeConnectDevice = device_dict[CONF_DEVICE]
+            for description in SWITCHES:
+                if description.on_key in hc_device.appliance.status:
+                    hc_switch = HomeConnectSwitch(
+                        device=hc_device,
+                        desc=description.key,
+                    )
+                    hc_switch.entity_description = description
```

Comment:
oh wait, HomeConnectSwitch `desc` parameter isn't the full parameter? If not, we should put the constructor bad, in that case I have overseen this. we should not set the entity_description outside